source/manual/alerts/rabbitmq-idle-consumers.html.md

```diff
@@ -0,0 +1,78 @@
+---
+owner_slack: "#govuk-2ndline"
+title: 'RabbitMQ: Consumers idle'
+parent: "/manual.html"
+layout: manual_layout
+section: Icinga alerts
+last_reviewed_on: 2020-08-24
+review_in: 6 months
+---
+
+This checks whether the RabbitMQ consumers for a queue have been active in the
+last 5 minutes.
+
+Publishing API [sends a heartbeat message][heartbeat_messages] every minute to
+the following queues - `email-alert-service`, `cache_clearing_service-high` and
+`content_data_api`. This should ensure that the consumers are never idle
+for these queues.
+
+## How this check works
+
+Icinga connects to RabbitMQ's admin API to check on the activity of the
+consumers of a given RabbitMQ message queue. This script for this check lives
+in [puppet][idle-puppet-config].
+
+## Troubleshooting steps
+
+### Check the RabbitMQ logs
+
+* Check the logs for the `rabbitmq` application which can be achieved using
+  [this Kibana query][kibana-query]. Is there evidence of any errors?
+
+### Check the RabbitmQ Grafana dashboard
```

Comment:
The content here is helpful. Somewhat minor: could we reorganise it so that it's easier to read through quickly? I don't think we care about the exchange graphs, so we've just got three things to look for:

- Check for high 'ready' messages for the queue
- Check for high unACK messages for the queue
- Check for high 'redeliver' rate for the queue

(The troubleshooting steps do seem to be roughly the same for each of these.)